wolfcrypt/src/tfm.c

```diff
@@ -1868,9 +1868,23 @@ void fp_montgomery_reduce(fp_int *a, fp_int *m, fp_digit mp)
 
 void fp_read_unsigned_bin(fp_int *a, const unsigned char *b, int c)
 {
+    /* must leave room for extra digit because of fp_mul_2d below */
+#if defined(ALT_ECC_SIZE) || defined(HAVE_WOLF_BIGINT)
+  const word32 maxC = (a->size - 1) * sizeof(fp_digit);
```

Comment:
What about the case that doesn't use fp_mul_2d?  That is, why are we shortening the max input in that case?